x/observer/keeper/msg_server_update_observer_test.go

```diff
@@ -0,0 +1,487 @@
+	})
+	t.Run("fail to update observer using regular account and update type admin", func(t *testing.T) {
+		k, ctx := keepertest.ObserverKeeper(t)
+		srv := keeper.NewMsgServerImpl(*k)
+
+		// #nosec G404 test purpose - weak randomness is not an issue here
+		r := rand.New(rand.NewSource(9))
+
+		// Set validator in the store
+		validator := sample.Validator(t, r)
+		validatorNew := sample.Validator(t, r)
+		validatorNew.Status = stakingtypes.Bonded
+		k.GetStakingKeeper().SetValidator(ctx, validatorNew)
+		k.GetStakingKeeper().SetValidator(ctx, validator)
+
+		consAddress, err := validator.GetConsAddr()
+		assert.NoError(t, err)
+		k.GetSlashingKeeper().SetValidatorSigningInfo(ctx, consAddress, slashingtypes.ValidatorSigningInfo{
+			Address:             consAddress.String(),
+			StartHeight:         0,
+			JailedUntil:         ctx.BlockHeader().Time.Add(1000000 * time.Second),
+			Tombstoned:          false,
+			MissedBlocksCounter: 1,
+		})
+
+		chains := k.GetParams(ctx).GetSupportedChains()
+		accAddressOfValidator, err := types.GetAccAddressFromOperatorAddress(validator.OperatorAddress)
+		assert.NoError(t, err)
+		count := uint64(0)
+		for _, chain := range chains {
+			k.SetObserverMapper(ctx, &types.ObserverMapper{
+				ObserverChain: chain,
+				ObserverList:  []string{accAddressOfValidator.String()},
+			})
+			count += 1
+		}
+
+		k.SetNodeAccount(ctx, types.NodeAccount{
+			Operator: accAddressOfValidator.String(),
+		})
+
+		newOperatorAddress, err := types.GetAccAddressFromOperatorAddress(validatorNew.OperatorAddress)
+		assert.NoError(t, err)
+
+		k.SetLastObserverCount(ctx, &types.LastObserverCount{
+			Count: count,
+		})
+
+		_, err = srv.UpdateObserver(sdk.WrapSDKContext(ctx), &types.MsgUpdateObserver{
+			Creator:            sample.AccAddress(),
+			OldObserverAddress: accAddressOfValidator.String(),
+			NewObserverAddress: newOperatorAddress.String(),
+			UpdateReason:       types.ObserverUpdateReason_AdminUpdate,
+		})
+		assert.ErrorIs(t, err, types.ErrUpdateObserver)
+	})
+}
+
+func TestUpdateObserverList(t *testing.T) {
+	t.Run("update observer list", func(t *testing.T) {
+		oldObserverAddress := sample.AccAddress()
+		newObserverAddress := sample.AccAddress()
+		list := []string{sample.AccAddress(), sample.AccAddress(), sample.AccAddress(), oldObserverAddress}
+		assert.Equal(t, oldObserverAddress, list[3])
+		keeper.UpdateObserverList(list, oldObserverAddress, newObserverAddress)
+		assert.Equal(t, 4, len(list))
+		assert.Equal(t, newObserverAddress, list[3])
+	})
+}
+
+func TestKeeper_UpdateObserverAddress(t *testing.T) {
+	t.Run("update observer address", func(t *testing.T) {
+		k, ctx := keepertest.ObserverKeeper(t)
+		oldObserverAddress := sample.AccAddress()
+		newObserverAddress := sample.AccAddress()
+		chains := k.GetParams(ctx).GetSupportedChains()
+		observerList := []string{sample.AccAddress(), sample.AccAddress(), sample.AccAddress(), oldObserverAddress}
```

Comment:
```suggestion
		observerList := []string{sample.AccAddress(), sample.AccAddress(), sample.AccAddress(), oldObserverAddress, sample.AccAddress(), sample.AccAddress()}
```
I think we can add some other addresses after to check it doesn't truncate or change those?